app/src/components/navigation/ProfileMenu.vue

```diff
@@ -1,25 +1,64 @@
 <script setup lang="ts">
 import { Menu, MenuButton, MenuItem, MenuItems } from "@headlessui/vue";
-import { ChevronDownIcon, UserIcon } from "@heroicons/vue/20/solid";
+import {
+    ChevronDownIcon,
+    UserIcon,
+    ArrowRightEndOnRectangleIcon,
+    ArrowLeftEndOnRectangleIcon,
+    Cog6ToothIcon,
+} from "@heroicons/vue/20/solid";
 import { useAuth0 } from "@auth0/auth0-vue";
 import { useRouter } from "vue-router";
+import { computed, ref } from "vue";
+import { LanguageIcon } from "@heroicons/vue/24/solid";
+import LanguageModal from "@/components/content/LanguageModal.vue";
+import { appLanguageAsRef } from "@/globalConfig";
 
-const { user, logout } = useAuth0();
+const { user, logout, isAuthenticated } = useAuth0();
 const router = useRouter();
+const showLanguageModal = ref(false);
 
-const userNavigation: {
-    name: string;
-    action: Function;
-}[] = [
-    { name: "Settings", action: () => router.push({ name: "settings" }) },
-    {
-        name: "Sign out",
-        action: async () => {
-            localStorage.removeItem("usedAuth0Connection");
-            await logout({ logoutParams: { returnTo: window.location.origin } });
+const commonNavigation = computed(() => {
+    return [
+        {
+            name: "Settings",
+            icon: Cog6ToothIcon,
+            action: () => router.push({ name: "settings" }),
         },
-    },
-];
+        {
+            name: "Language",
```

Comment:
I think a test for that it shows the modal when clicking the language button would be nice.